dango/cli/src/main.rs

```diff
@@ -93,25 +94,45 @@ async fn main() -> anyhow::Result<()> {
         });
 
         tracing_subscriber::registry()
+            .with(
+                tracing_subscriber::EnvFilter::try_from_default_env()
+                    .unwrap_or_else(|_| "info".into()), // Default to info if RUST_LOG not set
+            )
             .with(tracing_subscriber::fmt::layer().with_filter(filter))
             .with(sentry_layer())
             .init();
 
         tracing::info!("Sentry initialized");
     } else {
         tracing_subscriber::registry()
-            .with(tracing_subscriber::fmt::layer().with_filter(filter))
+            .with(
+                tracing_subscriber::EnvFilter::try_from_default_env()
+                    .unwrap_or_else(|_| "info".into()), // Default to info if RUST_LOG not set
+            )
+            .with(
+                tracing_subscriber::fmt::layer()
+                    // .with_span_events(tracing_subscriber::fmt::format::FmtSpan::CLOSE),
+            )
             .init();
     }
 
+    // Metrics should be initialized as soon as possible to capture all events.
+    let metrics_handle = init_metrics()?;
+
     match cli.command {
         Command::Db(cmd) => cmd.run(app_dir),
-        Command::Indexer(cmd) => cmd.run(app_dir).await,
+        Command::Indexer(cmd) => cmd.run(app_dir, metrics_handle).await,
         Command::Keys(cmd) => cmd.run(app_dir.keys_dir()),
         Command::Query(cmd) => cmd.run(app_dir).await,
-        Command::Start(cmd) => cmd.run(app_dir).await,
+        Command::Start(cmd) => cmd.run(app_dir, metrics_handle).await,
         #[cfg(feature = "testing")]
         Command::Test(cmd) => cmd.run().await,
         Command::Tx(cmd) => cmd.run(app_dir).await,
     }
```

Comment:
@penso It looks like only `Command::Indexer` and `Command::Start` need the `metrics_handle`. Do we need to init it even if `cli.command` is a different variant?

Reply:
We could probably init in in each subcommand if you'd like that.

Reply:
Can you see if this works: https://github.com/left-curve/left-curve/pull/817/commits/0772c5e0d9a3d27b2d62ee9ed77683bf0b3692d8